packages/loadbalancer/README.md

```diff
@@ -0,0 +1,201 @@
+# Load Balancer for WebRTC Mediasoup Servers
+
+## 🚀 Overview
+
+This project implements a **load balancer** for WebRTC **Mediasoup** servers. It enables **automatic server registration**, **health monitoring**, and **dynamic load balancing** to efficiently distribute WebRTC rooms among the available servers, ensuring a scalable and robust architecture.
+
+---
+
+## ⚡ **Key Features**
+
+### **Server Registration**
+
+- **Dynamic server registration**: Servers register themselves dynamically when they start up.
+- **Duplicate prevention**: If a server is already registered, the previous entry is removed and updated.
+- **Capacity calculation**: Each server's **total capacity** is computed based on the number of workers (`workers × 500 consumers`).
+- **Storage in Redis**: All registered servers are stored in Redis for fast retrieval.
+
+### **Health Monitoring & Auto-Removal**
+
+- **Scheduled health checks**: A background process continuously monitors all registered servers.
+- **HTTP-based ping test**: Each server is periodically tested via an HTTP `GET` request to the `/health` endpoint.
+- **Automatic deactivation**: If a server is unresponsive or returns a non-200 status code, it is marked as **unhealthy (`health: false`)**.
+- **Reactivation**: If a previously unhealthy server becomes responsive, it is marked as **healthy (`health: true`)**.
+
+### **Room Allocation & Load Balancing**
+
+- **Load-aware room allocation**: When creating a room, the load balancer selects the server with the **highest available capacity**.
+- **Capacity tracking**: Each time a room is created, the number of **expected consumers** is subtracted from the server’s capacity.
+- **Room deallocation**: When a room is closed, the consumers are freed, and the server’s available capacity is restored.
+
+### **Fault Tolerance & Scalability**
+
+- **Multi-instance support**: The system can be deployed in a **distributed** environment with multiple load balancer instances.
+- **Event-driven updates**: Server states and capacity updates are immediately reflected across all instances via Redis.
+- **Automatic failover**: If a WebRTC server becomes unavailable, new rooms are assigned only to healthy servers.
+
+---
+
+## 📌 **Endpoints**
+
+### ✅ **Register a WebRTC Server**
+
+Registers a new Mediasoup server, calculating its total capacity.
```

Comment:
What do you mean by "calculating"? Shouldn't just be "specifying its total capacity"?